src/core/rattle.cpp

```diff
@@ -170,16 +170,19 @@ void app_pos_correction(const ParticleRange &particles) {
   } // for i loop
 }
 
-void correct_pos_shake(const ParticleRange &particles) {
+void correct_pos_shake(ParticleRange const &particles) {
+  cells_update_ghosts();
   int repeat_, cnt = 0;
   int repeat = 1;
 
+  // Note: The particle range has to be re-obtained from the cell_structure
```

Comment:
This is not true, they are invalidated by a resort. It's enough to get them again after `cells_update_ghosts`, it's not needed after every ghost comm. Why would it?